source/ConditionAssignments.cpp

```diff
@@ -32,52 +50,168 @@ ConditionAssignments::ConditionAssignments(const DataNode &node)
 // Load a set of conditions from the children of this node.
 void ConditionAssignments::Load(const DataNode &node)
 {
-	setToEvaluate.Load(node);
+	if(!node.HasChildren())
+		node.PrintTrace("Error: Loading empty set of assignments");
+
+	// Loop through all children, and parse each line into an Assignment.
+	for(const DataNode &child : node)
+	{
+		Add(child);
+	}
 }
 
 
 
 // Save a set of conditions.
 void ConditionAssignments::Save(DataWriter &out) const
 {
-	setToEvaluate.Save(out);
+	for(const Assignment &assignment : assignments)
+	{
+		AssignOp aso = assignment.assignOperator;
+
+		auto it = find_if(ASSIGN_OP_TO_TEXT.begin(), ASSIGN_OP_TO_TEXT.end(),
+		[&aso](const std::pair<AssignOp, const string> &e) {
+			return e.first == aso;
+		});
+		if(it != ASSIGN_OP_TO_TEXT.end())
+		{
+			out.WriteToken(assignment.conditionToAssignTo);
+			out.WriteToken(it->second);
+			assignment.expressionToEvaluate.SaveSubset(out);
+			out.Write();
+		}
+	}
 }
 
 
 
 // Check if there are any entries in this set.
 bool ConditionAssignments::IsEmpty() const
 {
-	return setToEvaluate.IsEmpty();
+	return assignments.empty();
 }
 
 
 
 // Modify the given set of conditions.
 void ConditionAssignments::Apply(ConditionsStore &conditions) const
 {
-	for(const ConditionSet::Expression &expression : setToEvaluate.expressions)
-		if(!expression.IsTestable())
-			expression.Apply(conditions);
+	for(const Assignment &assignment : assignments)
+	{
+		auto &ce = conditions[assignment.conditionToAssignTo];
+		int64_t newValue = assignment.expressionToEvaluate.Evaluate(conditions);
+		switch(assignment.assignOperator)
+		{
+			case AO_ASSIGN:
+				ce = newValue;
+				break;
+			case AO_ADD:
+				ce += newValue;
+				break;
+			case AO_SUB:
+				ce -= newValue;
+				break;
+			case AO_MUL:
+				ce = static_cast<int64_t>(ce) * newValue;
+				break;
+			case AO_DIV:
+				// TODO: should be crash-safe implemented on Condition Entry:
+				ce = static_cast<int64_t>(ce) / newValue;
```

Comment:
```suggestion
				ce = newValue ? static_cast<int64_t>(ce) / newValue : numeric_limits<int64_t>::max();
```
Based on ConditionSet's implementation